### ISSUE TITLE
.NET Core 1.1.1 runtime exe installer has unsigned files

### ISSUE THREAD

**Issue by jmarolf** [dotnet/core-setup#2082]
See the following internal bugs:
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/408746
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/408748
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/409061
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/409063

These scans are looking at both the [x86](https://go.microsoft.com/fwlink/?linkid=843437) and [x64](https://go.microsoft.com/fwlink/?linkid=843433) installers and finding file with fingerprint 07A9E58F4D0FCFE7C1BCFBDABB1742E4 to be unsigned.

**Comment by gkhanna79** [comment]
@mikem8361 @dagood Please look into this for 2.0 and also for 1.0.X and 1.1.x.

CC @leecow @lt72

**Comment by gkhanna79** [comment]
What are the next steps for this? Is this issue present for 1.1.2/1.0.5 and Preview1 @mikem8361 and @dagood ?

**Comment by mikem8361** [comment]
I have no idea how to fix this. I assumed the build system in coreclr would take care of it. 

@weshaggard I thought SOS.NETCore was going to be moved to the corefx repo?

**Comment by gkhanna79** [comment]
@mikem8361 Please work with @dagood to figure out the resolution for this and determine the impacted releases. I would suggest starting by confirming if master and release/2.0.0 branches are affected or not.

**Comment by weshaggard** [comment]
>@weshaggard I thought SOS.NETCore was going to be moved to the corefx repo?

That has definitely been talked about for the build form source effort in order to eliminate the repo cycle but there hasn't been any planned work to do it (yet).

**Comment by jmarolf** [comment]
@gkhanna79 for reasons not related to this bug I am removing the setup authoring that pulls in these installers https://devdiv.visualstudio.com/DevDiv/_git/VS/pullrequest/65511 This should unblock devdiv from shipping for the time being.

**Comment by gkhanna79** [comment]
Can you please elaborate on the reason and what does it specifically unblock?

**Comment by jmarolf** [comment]
@gkhanna79 this all started when I checked in setup authoring for Visual Studio that allowed users to choose to install older CoreClr versions (1.0 and 1.1) in addition to 2.0 if they wanted.  This pulled in the  installers from https://www.microsoft.com/net/download/core#/runtime. As part of the Visual Studio build they check to see if everything that is being referenced is signed.  Currently because VS references these installers this is a ship-blocker for them until this gets sorted out.  Removing the setup authoring allows VS to ship, technically resolving the bugs but not the underlying issue.  We obviously still want to figure out why SOS is not signed and determine if that is desirable or not, its just that VS is no longer blocked from shipping.

**Comment by gkhanna79** [comment]
So, are you saying that 2.0 binaries are signed?

Also, was this setup authoring (to including 1.0 and 1.1) recent or has been there since day 1? 

CC @leecow incase this impacts any releases and @dagood and @ravimeda from the signing perspective.

**Comment by jmarolf** [comment]
@gkhanna79 I cannot state if 2.0 binaries are signed.  I do not think there is any setup authoring that deploys them today.  this setup authoring to including 1.0 and 1.1 runtime installers was very recent ~ last 2wks.

**Comment by mikem8361** [comment]
@dagood @gkhanna79 who knows how/where SOS.NETCore.dll is supposed to get signed? In the coreclr repo where it is built or is it part of core-setup? System.Private.CoreLib.dll is built in coreclr and it is signed. When I added SOS.NETCore I assumed it would be signed like that one. I need some help with this issue.

**Comment by mikem8361** [comment]
I noticed the SOS.NETCore.csproj doesn't have these properties and System.Private.CoreLib.csproj does.

<SignAssembly>true</SignAssembly>
<DelaySign>true</DelaySign>

Would adding them enable signing?  How do I tell if an assembly is properly signed?

**Comment by gkhanna79** [comment]
@weshaggard Do you have thoughts on this?

**Comment by weshaggard** [comment]
Not sure without digging into the project in this repo. In general for corefx this is handled for projects because they automatically (indirectly) import https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/sign.targets.

Even if you set those properties you need to make sure you give it a key and you also need to make sure it is setup to get real signed. For this new assembly I would suggest you use the open key which is what we are using for all new assemblies.

**Comment by mikem8361** [comment]
It looks like in the coreclr repo build files "sign.targets" is included by SOS.NETCore.csproj (indirectly).  The
property:
`    <UseOpenKey Condition="'$(UseOpenKey)'==''">true</UseOpenKey>
`
is already in the .csproj file so it looks adding the above properties should do it. Does anybody know if my local coreclr build will signed the binaries? Or does it have to be an "official" build for some reason?

**Comment by mikem8361** [comment]
@jmarolf Is only strong name signing is required for SOS.NETCore.dll and not Authenticode?

**Comment by danmosemsft** [comment]
Per @gkhanna79 , this is a dupe of #1801
@crummel

**Comment by jmarolf** [comment]
@danmosemsft the underlying issue may be fixed by #1801 but the installers listed on https://www.microsoft.com/net/download/core#/runtime still contain unsigned binaries.  To fix this issue new installers will need to be generated for 1.0 and 1.1 that have this fix

**Comment by jmarolf** [comment]
@danmosemsft I think the intention was to have #1801 track the work of generating the installers? Closing in that case

**Comment by dagood** [comment]
I think this isn't quite a dupe, but still should be closed: I think it's resolved by https://github.com/dotnet/core-setup/pull/2221 that was applied to `release/1.1.0`, and generating the installers sounds to me like what we're planning on doing by releasing 1.1.2. (I don't know if that needs a specific new issue.)

#1801 looks to me like it needs (or already has) a specific `master` and/or `release/2.0.0` fix.

**Comment by jmarolf** [comment]
@dagood well as long as new installers for 1.0 and 1.1 (I assume 1.0.5 and 1.1.2) get released with this fix I am fine with it.
